### PR TITLE
Write signed manifests to the CBOR format.

### PR DESCRIPTION
**Pull request by jyasskin**
This sits on top of #55, #60, #61, #63, #64, #65, and #66. Sorry that it's so long again.

### CODE REVIEW

File: go/webpack/cbor_format.go [hunk 1]
```diff
@@ -108,29 +156,172 @@ func WriteCBOR(p *Package, to io.Writer) error {
...
+		metadataKeys = append(metadataKeys, key)
+	}
+	sort.Slice(metadataKeys, func(i, j int) bool {
+		return cbor.CanonicalLessStrings(metadataKeys[i], metadataKeys[j])
+	})
+	metadata := manifest.AppendMap(uint64(len(metadataKeys)))
+	for _, key := range metadataKeys {
+		metadata.AppendUTF8S(key)
+		switch key {
+		case "origin":
```
Comment (mrdewitt) [inline go/webpack/cbor_format.go]: micro nit: process "date" first to match the order in the spec

File: go/webpack/cbor_format_test.go [hunk 2]
```diff
@@ -27,6 +42,17 @@ func hpackByteArray(headersAndValues ...string) []byte {
 	return append(cbor.Encoded(cbor.TypeBytes, len(result)), result...)
 }
 
+func hpackDecode(t *testing.T, encoded []byte) HTTPHeaders {
```
Comment (mrdewitt) [inline go/webpack/cbor_format_test.go]: is it more canonical to name the return value than to declare one in the first line of the function?

File: go/webpack/cbor_format_test.go [hunk 3]
```diff
@@ -88,3 +114,209 @@ func TestWriteCBOR(t *testing.T) {
...
+	assert.EqualValues("certificates", d.read(d.decodeType(cbor.TypeText)))
+	numCertificates := d.decodeType(cbor.TypeArray)
+	certs := make([]*x509.Certificate, numCertificates)
+	for i := range certs {
+		certBytes := d.read(d.decodeType(cbor.TypeBytes))
+		certs[i], err = x509.ParseCertificate(certBytes)
+		require.NoError(err)
+	}
+	signingCert := certs[keyIndex]
+	signingCert.Verify(x509.VerifyOptions{
```
Comment (mrdewitt) [inline go/webpack/cbor_format_test.go]: Does this actually verify that signingCert corresponds to a DNSName of "example.com"?  Does it make sense to add a test for attempts to sign a package with no signatures corresponding to the origin?

File: go/webpack/cbor_format.go (hunk 1 above)
Comment (jyasskin) [inline go/webpack/cbor_format.go]: Done.

File: go/webpack/cbor_format_test.go (hunk 2 above)
Comment (jyasskin) [inline go/webpack/cbor_format_test.go]: I don't know, but I can do that instead.

File: go/webpack/cbor_format_test.go (hunk 3 above)
Comment (jyasskin) [inline go/webpack/cbor_format_test.go]: Yes, and I've added an extra test with the wrong DNSName to check that.

And I've added a check that at least one signature covers the package's origin.

**Pull request closed without merging**